stl/inc/random

```diff
@@ -1722,8 +1721,9 @@ public:
     friend basic_istream<_Elem, _Traits>& operator>>(
         basic_istream<_Elem, _Traits>& _Istr, shuffle_order_engine& _Eng) { // read state from _Istr
         _Istr >> _Eng._Eng;
-        for (size_t _Idx = 0; _Idx < shuffle_order_engine::table_size; ++_Idx) {
-            _Istr >> _Eng._Arr[_Idx];
+
+        for (auto& _Val : _Eng._Arr) {
+            _Istr >> _Val;

```

Comment:
Is the blank line *before* the `for` loop a rule, or just something you preferred in this particular case (and line 1736)? Just wondering if it's something I should try to follow going forward or not.

Reply:
Thanks for asking! Not a rule, just a preference. Having blank lines between different "phases" of work seems more readable to me. In these cases, I think I was also influenced by how the `return` statement performs a third phase of work - it seemed more symmetric to have a blank line on both sides of the `for` loop. (I might also have been looking at existing examples elsewhere like right line 4427.)

Sometimes, lines of code are needed to prepare for a loop, in which case having them directly next to the loop seems reasonable - like right lines 2224-2225.

But in either case I don't think I'd complain about the opposite pattern of blank lines. The only one I feel strongly about is newlines between non-chained `if` statements. I generally think that `if` mixed with `for` loops (or switches, etc.) should also have blank lines, and I do that in my own code, although I generally avoid requesting such changes in code reviews.

Reply:
Understood---thanks for taking the time to respond!